### ISSUE TITLE
Porting the rest of arcade.geometry

### ISSUE THREAD

**Issue by Cleptomania** [pythonarcade/arcade-accelerate#7]
Similar to arcade.math, arcade.geometry has some remaining functions which should be easy to port, and will aid in getting code into rust for use by bigger systems later.

Below is a check list of the remaining functions:

- [x] are_polygons_intersecting
- [x] is_point_in_box
- [x] get_triangle_orientation
- [x] are_lines_intersecting(dependent on get_triangle_orientation)
- [x] is_point_in_polygon

**Comment by jf22201** [comment]
Hi, I'm trying to get started with contributing to OSS, I was wondering if I could try and work on this?

**Comment by codeguru42** [comment]
I'm working on `is_point_in_polygon()`.

**Comment by Cleptomania** [comment]
This is finished and all tests completed as of #18.

We also made some changes to the math of this due to the `%` operator in Rust. In Python the operator performs the modulus function, but in Rust it performs the remainder.

Instead of using the modulus operator in Rust, in order to match the intended functionality of the Python implementation we made use of the `rem_euclidian` function in the Rust standard library.
